### PR TITLE
Add project parameter to os_network

### PR DESCRIPTION
**Pull request by rcarrillocruz**
##### Issue Type:
- Feature Pull Request
##### Plugin Name:

os_network
##### Summary:

A cloud/domain admin should be able to create a network on any project
it is granted to.
This changes adds the possibility to pass either a project ID or
project name.
##### Example:

```
(venv) ubuntu@devstack:~$ cat test_os_network.yml 

---
- hosts: localhost
  connection: local
  gather_facts: false
  tasks:
  - os_network:
      cloud: devstack-cloud_admin
      state: present
      name: openstackci_net
      project: 7d517c389898420d8113d557357914eb
```

```

(venv) ubuntu@devstack:~$ ansible-playbook -i inventory test_os_network.yml -e "@infra_config.yml" -vvv
No config file found; using defaults

PLAYBOOK: test_os_network.yml **************************************************
1 plays in test_os_network.yml

PLAY [localhost] ***************************************************************

TASK [os_network] **************************************************************
task path: /home/ubuntu/test_os_network.yml:6
ESTABLISH LOCAL CONNECTION FOR USER: ubuntu
localhost EXEC /bin/sh -c '( umask 22 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1457953009.92-163528194082886 `" && echo "` echo $HOME/.ansible/tmp/ansible-tmp-1457953009.92-163528194082886 `" )'
localhost PUT /tmp/tmpZArkV_ TO /home/ubuntu/.ansible/tmp/ansible-tmp-1457953009.92-163528194082886/os_network
localhost EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/env python /home/ubuntu/.ansible/tmp/ansible-tmp-1457953009.92-163528194082886/os_network'
changed: [localhost] => {"changed": true, "id": "1f899f8e-0f28-4d6a-9c13-cf4eac9fcca1", "invocation": {"module_args": {"admin_state_up": true, "api_timeout": null, "auth": null, "auth_type": null, "availability_zone": null, "cacert": null, "cert": null, "cloud": "devstack-cloud_admin", "endpoint_type": "public", "external": false, "key": null, "name": "openstackci_net", "region_name": null, "shared": false, "state": "present", "timeout": 180, "verify": true, "wait": true}, "module_name": "os_network"}, "network": {"admin_state_up": true, "availability_zone_hints": [], "availability_zones": [], "id": "1f899f8e-0f28-4d6a-9c13-cf4eac9fcca1", "ipv4_address_scope": null, "ipv6_address_scope": null, "mtu": 1450, "name": "openstackci_net", "port_security_enabled": true, "provider:network_type": "vxlan", "provider:physical_network": null, "provider:segmentation_id": 1089, "router:external": false, "shared": false, "status": "ACTIVE", "subnets": [], "tenant_id": "7d517c389898420d8113d557357914eb"}}

PLAY RECAP *********************************************************************
localhost                  : ok=1    changed=1    unreachable=0    failed=0
```

### CODE REVIEW

**Comment by Shrews** [comment]
We need to begin adding some version checking to our modules. Take a look at how os_ironic_node.py uses StrictVersion and use that pattern in this module. We'll obviously need a shade release with your project_id change before you can check the version.

**Comment by mgale** [comment]
Does the value for project need to be the ID or can it be the project_name?

The reason I ask is that os_project currently does not allow the list of projects and can only update / create from my understanding. So getting the project ID might be difficult at this time, unless os_project_facts is also being developed currently?

**Comment by rcarrillocruz** [comment]
@mgale It can be project name, but the user authenticated must be a cloud/domain admin.

I agree having an os_project_facts would be useful, we should have os_<resource>_facts modules for each one of them, I can take a stab at it.

**Comment by rcarrillocruz** [comment]
@Shrews i added the shade version checking, thanks

**Comment by rcarrillocruz** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @rcarrillocruz. To the current maintainers, @emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks @rcarrillocruz for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by rcarrillocruz** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @rcarrillocruz. To the current maintainers, @emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/openstack/os_network.py [hunk 1]
```diff
@@ -182,10 +193,17 @@ def main():
     provider_physical_network = module.params['provider_physical_network']
     provider_network_type = module.params['provider_network_type']
     provider_segmentation_id = module.params['provider_segmentation_id']
+    project = module.params.pop('project')
 
     try:
         cloud = shade.openstack_cloud(**module.params)
-        net = cloud.get_network(name)
+        if project is not None:
+            project_id = cloud.get_project(project)['id']
```
Comment (Shrews) [inline cloud/openstack/os_network.py]: This is going to throw an exception if the project name/id is invalid and you attempt to get the 'id' key for a None value. You should fail_json() if the project is invalid here.

**Comment by rcarrillocruz** [comment]
ready_for_review

**Pull request merged**